crates/astria-composer/src/searcher/mod.rs

```diff
@@ -79,41 +77,13 @@ pub(super) struct Searcher {
 
 #[derive(Debug, Default)]
 pub(crate) struct Status {
-    eth_connected: bool,
+    all_collectors_connected: bool,
```

Comment:
thoughts on having this be a count of connected collectors rather than a boolean?

my thinking is that a composer is still healthy if only 1 of n collectors is working - but i'm not sure about the distinction between how `healthz` and `readyz` are used?

my rational here is along the lines that you wouldnt want a single rollup failing to disable your composer because then it would impact the other rollups as well.

Reply:
Right now this only flips to true once all requested providers respond.

I think this is reasonable for readyz because not having this indicates a problem.

I agree that for an actual status api request or for healthz this should look different.

We can also implement a config to require one or all rollups to be present.

But the current implementation is a direct extension of the previous functionality, so I think this would be out of scope.

Happy to get follow-up PRs though or to discuss how to do this.

Reply:
sgtm - will make an issue about this